lib/pure/terminal.nim

```diff
@@ -768,6 +768,43 @@ proc resetAttributes*() {.noconv.} =
   ## ``system.addQuitProc(resetAttributes)``.
   resetAttributes(stdout)
 
+proc restoreOnInterrupt() {.noconv.} =
+  ## called in `getPassword` if readLine is interrupted by SIGINT to remove the
+  ## typed characters and restore the terminal to default settings.
+  # erase line, to remove the password
+  stdout.eraseLine()
+  # restore default terminal settings
+  stdout.showCursor()
+  stdout.resetAttributes()
+  stdout.write("\n")
+  if stackTraceAvailable() == true:
```

Comment:
Is there a function that does this already which could be called? Presumably there is already a hook somewhere which does this.

P.S. ``== true`` is redundant.

Reply:
As far as I can tell there's no proc to do that. `resetAttributes` is the closest, but it only affects the text style and leaves the cursor as it is. One could of course define a `resetTerminal` proc, which combines the two.

The newline I insert should not be part of that proc though, since I only need it due to moving the cursor one line up to get back down. 

Regarding the if statement. Oh, good point, thanks. I'll change that.